LiteCore/Support/LogDecoder.cc

```diff
@@ -53,33 +57,26 @@ namespace litecore {
 
 
     void LogIterator::writeTimestamp(Timestamp t, ostream &out) {
-        struct tm tm;
-        localtime_r(&t.secs, &tm);
-        char timestamp[100];
-        strftime(timestamp, sizeof(timestamp), "%T", &tm);
-        out << timestamp;
-        sprintf(timestamp, ".%06u| ", t.microsecs);
-        out << timestamp;
+        local_time<microseconds> tp { seconds(t.secs) + microseconds(t.microsecs) };
+        struct tm tmpTime = FromTimestamp(duration_cast<seconds>(tp.time_since_epoch()));
+        tp -= GetLocalTZOffset(&tmpTime, true);
+        out << format("%T| ", tp);
     }
 
 
     void LogIterator::writeISO8601DateTime(Timestamp t, std::ostream &out) {
-        struct tm tm;
-        gmtime_r(&t.secs, &tm);
-        char timestamp[100];
-        strftime(timestamp, sizeof(timestamp), "%FT%T", &tm);
-        out << timestamp;
-        sprintf(timestamp, ".%06uZ", t.microsecs);
-        out << timestamp;
+        sys_time<microseconds> tp(seconds(t.secs) + microseconds(t.microsecs));
+        out << format("%FT%TZ", tp);
     }
 
 
     string LogIterator::formatDate(Timestamp t) {
-        struct tm tm;
-        localtime_r(&t.secs, &tm);
-        char timestamp[100];
-        strftime(timestamp, sizeof(timestamp), "%c", &tm);
-        return timestamp;
+        local_time<microseconds> tp(seconds(t.secs) + microseconds(t.microsecs));
+        struct tm tmpTime = FromTimestamp(duration_cast<seconds>(tp.time_since_epoch()));
+        tp -= GetLocalTZOffset(&tmpTime, true);
```

Comment:
Just curious why we use local time, not utc?

Reply:
I don't know the answer to that.